src/Program.ts

```diff
@@ -680,6 +772,14 @@ export class Program {
                 }
             }
 
+            // Build component types for any component that changes
+            this.logger.time(LogLevel.info, ['Build component types'], () => {
+                for (let { componentKey, componentName } of this.componentSymbolsToUpdate) {
+                    this.updateComponentSymbolInGlobalScope(componentKey, componentName);
+                }
+                this.componentSymbolsToUpdate.clear();
+            });
+
```

Comment:
It needs to be after the files are validated because then when we get a merged symbol table for the scope, everything that CAN be resolved will be able to be. 

That will be important for when we "leak" pocket types (for params/returns) between scopes via callFuncs, because we can't leak a ReferenceType (because those are ONLY valid in a linked scope).

Reply:
okay, sounds good.